File: Day 03/Reverse_Array.py
class ReverseArray:
    
    # Method to reverse an array in-place
    @staticmethod
    def reverse_array(arr):
        start = 0
        end = len(arr) - 1
        
        while start < end:
            # Swap elements
            arr[start], arr[end] = arr[end], arr[start]
            
            start += 1
            end -= 1
    
    # Method to print array
    @staticmethod
    def print_array(arr):
        for element in arr:
            print(element, end=" ")
        print()  # New line
    
    @staticmethod
    def main():
        arr = [1, 2, 3, 4, 5, 6]
        
        print("Original array:")
        ReverseArray.print_array(arr)
        
        ReverseArray.reverse_array(arr)
        
        print("Reversed array:")
        ReverseArray.print_array(arr)

# Run the main method
if __name__ == "__main__":
    ReverseArray.main()

File: Day 03/ReverseArray.java
public class ReverseArray {
    
    // Method to reverse an array in-place
    public static void reverseArray(int[] arr) {
        int start = 0;
        int end = arr.length - 1;
        
        while (start < end) {
            // Swap elements
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            
            start++;
            end--;
        }
    }
    
    // Method to print array
    public static void printArray(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    
    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5, 6};
        
        System.out.println("Original array:");
        printArray(arr);
        
        reverseArray(arr);
        
        System.out.println("Reversed array:");
        printArray(arr);
    }
}